system-tests/junit5-execution-listener-test/build.gradle

```diff
@@ -0,0 +1,31 @@
+plugins {
+    id 'com.teamscale.java-convention'
+    id 'com.teamscale.coverage'
+}
+
+test {
+    def agentJar = project(':agent').tasks.shadowJar.outputs.files.files.first()
+    environment("AGENT_PATH", agentJar)
+    environment("AGENT_VERSION", version)
+}
+
+dependencies {
+    testImplementation project(':tia-client')
+    testImplementation project(':common-system-test')
+
+    testImplementation 'com.sparkjava:spark-core:2.9.3'
+    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.7.2'
+    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.7.2'
+    testImplementation 'org.assertj:assertj-core:3.19.0'
+    testImplementation 'com.squareup.moshi:moshi:1.12.0'
+    testImplementation 'com.teamscale:teamscale-lib-commons:7.0.2'
+}
+
+tasks.test.dependsOn(project(':agent').tasks.shadowJar)
+// install dependencies needed by the Maven test projects
+rootProject.getAllprojects().forEach {
+    def publishTask = it.tasks.findByPath("publishToMavenLocal")
+    if (publishTask != null) {
+        tasks.test.dependsOn(publishTask)
+    }
+}
```

Comment:
The current `iGP` task does not publish all projects but only a manually maintained subset, namely:

```groovy
		':impacted-test-engine:publishToMavenLocal',
		':report-generator:publishToMavenLocal',
		':teamscale-gradle-plugin:publishToMavenLocal',
		':teamscale-client:publishToMavenLocal',
		':tia-client:publishToMavenLocal'
```

However, it's not clear to me what determines, which projects should be on that list. Can you comment? Or should I change this to programmatically publish all subprojects?

One other weird thing: running `./gradlew publishToMaven` already works (even though there is no such task in the root project) and publishes all subprojects. However, I can't depend on that (probably since it's not a real task but just some command line fu for convenience that delegates to subprojects). Or am I missing something?

Reply:
Yeah, `./gradlew publishToMaven` is just CLI convenience and collects all tasks with that name. 

And you are right about the subset: I missed that. The list holds everything but the agent because it took too long to build it everytime, but AFAIR this is cacheable by now so making it depend on all publishToMavenLocal should be fine by now. Then we can also get rid of the manually maintained list